php/helpers.php

```diff
@@ -309,3 +309,136 @@ function block_field_config( $name ) {
 	}
 	return (array) $block_lab_config->fields[ $name ];
 }
+
+/**
+ * Add a new block.
+ *
+ * @param string $block_name   The block name (slug), like 'example-block'.
+ * @param array  $block_config {
+ *     An associative array containing the block configuration.
+ *
+ *     @type string   $title    The block title.
+ *     @type string   $icon     The block icon. See assets/icons.json for a JSON array of all possible values. Default: 'block_lab'.
+ *     @type string   $category The slug of a registered category. Categories include: common, formatting, layout, widgets, embed. Default: 'common'.
+ *     @type array    $excluded Exclude the block in these post types. Default: [].
+ *     @type string[] $keywords An array of up to three keywords. Default: [].
+ *     @type array    $fields {
+ *         An associative array containing block fields. Each key in the array should be the field slug.
+ *
+ *         @type array {$slug} {
+ *             An associative array describing a field. Refer to the $field_config parameter of block_lab_add_field().
+ *         }
+ *     }
+ * }
+ */
+function block_lab_add_block( $block_name, $block_config = array() ) {
+	$block_config['name'] = str_replace( '_', '-', sanitize_title( $block_name ) );
+
+	if ( ! isset( $block_config['title'] ) ) {
+		$block_config['title'] = ucwords( $block_config['name'], '-' );
+		$block_config['title'] = str_replace( '-', ' ', $block_config['title'] );
+	}
+
+	if ( ! isset( $block_config['icon'] ) ) {
+		$block_config['icon'] = 'block_lab';
+	}
+
+	if ( ! isset( $block_config['category'] ) ) {
+		$block_config['category'] = 'common';
+	}
+
+	if ( ! isset( $block_config['excluded'] ) ) {
+		$block_config['excluded'] = array();
+	}
+
+	if ( ! isset( $block_config['keywords'] ) ) {
+		$block_config['keywords'] = array();
+	}
+
+	if ( ! isset( $block_config['fields'] ) ) {
+		$block_config['fields'] = array();
+	}
+
+	add_filter(
```

Comment:
The logic in the callback looks good.

What do you think about adding a public function to class `Loader` like:

`block_lab()->loader->add_block( $block_config );`

That could have most of the logic in this callback, but this particular `add_filter()` call wouldn't be needed.

The `apply_filters( 'block_lab_blocks' )` call in `Loader` would still be useful, as would this `block_lab_add_block()` function.

Reply:
Addressed in https://github.com/getblocklab/block-lab/pull/434/commits/5b98f71fd97a3e6fb6973b89c98f15bbb1993e10.